File: static/code/sample/tethered/gdl-sospectrum-a.py
# In this sample:
# Get 7 channel data from Spectrum and show it on the console

import time
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
while True:
    data = bytearray(7)  # Python equivalent of byte[7]
    
    duelink.Engine.ExecuteCommand("UpdBnd()")
    duelink.Stream.ReadBytes("b1", data)
    
    print("============================")
    for i in range(7):
        print(f"data {i}: {data[i]}")
    
    time.sleep(0.001)  # Sleep 1 millisecond

File: static/code/sample/tethered/gdl-sospectrum-a.cs
// In this sample:
// Get 7 channel data from Spectrum and show it on the console

using System;
using System.Drawing;
using GHIElectronics.DUELink;
using static System.Net.Mime.MediaTypeNames;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

while (true) {
    var data = new byte[7];
   
    duelink.Engine.ExecuteCommand("UpdBnd()");
    duelink.Stream.ReadBytes("b1",data);
    Console.WriteLine("============================");
    for (var i = 0; i < 7; i++) {
        Console.WriteLine($"data {i}: {data[i]}");        
    }    
    Thread.Sleep(1);
}
